week3/homework/src/actions.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { readFile, writeFile } = require('fs');
+const { promisify } = require('util');
+const uuid = require('uuid/v4');
+const read = promisify(readFile);
+const write = promisify(writeFile);
+const FILE_PATH = 'todo.json';
+const list = {};
+
+
+const fileContent = (todo) => {
+    return write(FILE_PATH, JSON.stringify(todo, null, 2));
+};
+
+const createTodo = async (todo) => {
+    if (!todo.description) throw new Error('Description is required');
+
+    const todoList = await readTheList();
+    const id = uuid();
+    list[id] = todo;
+    list[id].done = false;
+    todoList.push(list);
+    await fileContent(todoList);
+    return todoList;
+};
+
+const readTheList = () => {
+    return read(FILE_PATH, 'utf-8')
+        .then(JSON.parse)
+        .catch(() => ([]));
+};
+
+const updateTodo = async (id, newTodo) => {
+    const todoList = await readTheList();
+    try {
+        const foundItem = todoList.find(todo => todo[id]);
+        foundItem[id].description = newTodo.description;
+        await fileContent(todoList);
+        return todoList;
+    } catch (error) {
+        throw new Error('Item cannot be found.');
+    }
+};
+
+const markAsDoneOrNotDone = async (id, boolean) => {
+    const todoList = await readTheList();
+
+    try {
+        const foundItem = todoList.find(todo => todo[id]);
+        foundItem[id].done = boolean;
+        await fileContent(todoList);
+        return todoList;
+    } catch (error) {
+        throw new Error('Item cannot be found.')
+    }
+};
+
+const clearTodos = async () => {
+    const todoList = await readTheList();
+
+    if (todoList.length === 0) throw new Error('The list is empty');
```

Comment:
What is the purpose of this check?

Reply:
oh, it should not be there, I was supposed to pass it in,  delete by id, function like I did in command line app version :sweat_smile:  but I guess I don't need that here because a user would first get the list and then delete based on id. And if the list was empty they would see it, so as I understand this check is not necessary.